src/main/java/com/sprint/mission/discodeit/repository/file/FileUserRepository.java

```diff
@@ -0,0 +1,61 @@
+package com.sprint.mission.discodeit.repository.file;
+
+import com.sprint.mission.discodeit.entity.User;
+import com.sprint.mission.discodeit.repository.UserRepository;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Repository;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+@ConditionalOnProperty(name = "discodeit.repository.type", havingValue = "file")
+@Repository
+public class FileUserRepository implements UserRepository {
+    private static final Path directory = Paths.get("./src/main/resources/UserDate");
+    private static final FileEdit instance = new  FileEdit();
+
+    private Path filePaths(UUID id) {
+        return directory.resolve( id + ".ser");
+    }
+
+    public FileUserRepository() {
+        instance.init(directory);
+    }
+
+    @Override
+    public User save(User user) {
+        instance.save(directory,user.getId(),user);
+        return user;
+    }
+
+    @Override
+    public Optional<User> findById(UUID id) {return instance.load(directory,id);}
+
+    @Override
+    public Optional<User> findByEmail(String email) {
+        List<User> user = instance.loadAll(directory);
```

Comment:
복수이므로 `users`가 변수명에 좀더 적절할 것 같아요.
```suggestion
        List<User> users = instance.loadAll(directory);
```